iib/web/models.py

```diff
@@ -360,6 +370,15 @@ def add_state(self, state, state_reason):
         db.session.flush()
         self.request_state_id = request_state.id
 
+    def add_build_tag(self, name):
```

Comment:
Also, shouldn't we be checking if a build tag already exists before adding a new one associated with the request? Check out `add_architecture` for inspiration.

Reply:
Yeah, we should. I thought sqlalchemy is smart enough to figure this out